cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -135,6 +136,61 @@ Result<BatchesWithSchema> MakeBatchesFromNumString(
   return batches;
 }
 
+
+// Can handle dictionary fields in schema
+Result<BatchesWithSchema> MakeBatchesFromNumString(
+    const std::shared_ptr<Schema>& schema,
+    const std::vector<std::string_view>& json_strings, int multiplicity = 1) {
+  // Create a simple schema with the value type in the place of any dictionary types
+  bool has_dict_columns = false;
+  FieldVector simple_fields;
+  for (auto field : schema->fields()) {
+    if (field->type()->id() == Type::DICTIONARY) {
+      has_dict_columns = true;
+      auto value_type =
+          checked_cast<const DictionaryType&>(*field->type()).value_type();
+      simple_fields.push_back(field->WithType(value_type));
+    } else {
+      simple_fields.push_back(field);
+    }
+  }
+
+  // If there are no dictionary columns, we can use the simpler implementation
+  if (!has_dict_columns) {
+    return MakeBatchesFromNumStringImpl(schema, json_strings, multiplicity);
+  }
+
+  // Otherwise, we create batches using the simple schema
+  auto simple_schema =
+      std::make_shared<Schema>(simple_fields, schema->endianness(), schema->metadata());
+  ARROW_ASSIGN_OR_RAISE(auto simple_batches,
+      MakeBatchesFromNumStringImpl(simple_schema, json_strings, multiplicity));
+  
+  // and dictionary-encode the columns that were originally dictionaries
+  // TODO: For the real implementation, each column should share a dictionary, so
```

Comment:
nit: Probably a stale comment; I think we ended up deciding that this doesn't matter

Reply:
Actually I stand corrected as per https://docs.google.com/document/d/1hsOJtlWVm5j78V4Etbt7IqsNXxWYmPFGoist2TrI7Cw/edit?tab=t.0; please ignore!